src/backend/distributed/executor/multi_real_time_executor.c

```diff
@@ -107,7 +109,8 @@ MultiRealTimeExecute(Job *job)
 	}
 
 	/* loop around until all tasks complete, one task fails, or user cancels */
-	while (!(allTasksCompleted || taskFailed || QueryCancelPending))
+	while (!(allTasksCompleted || taskFailed || QueryCancelPending ||
+			 sizeLimitIsExceeded))
```

Comment:
For the sake of consistency, I want to keep this check in while && (add `continue;` instead of `break;` || just remove `break;`) 
Which one do you suggest?

Reply:
I like `break;` because it makes it obvious that we immediately leave the loop. Could also live with just leaving both the `break` and the `|| sizeLimitIsExceeded` here.